blocksync/reactor_test.go

```diff
@@ -42,10 +42,13 @@ func randGenesisDoc(numValidators int, randPower bool, minPower int64) (*types.G
 	}
 	sort.Sort(types.PrivValidatorsByAddress(privValidators))
 
+	consPar := types.DefaultConsensusParams()
+	consPar.ABCI.VoteExtensionsEnableHeight = 1
```

Comment:
Is there a test to check going from extensions disabled to extensions enabled?
If so, this could be a parameter to allow creating genesis of both kinds.

Reply:
Yes, there are some, but they are elsewhere.
This PR's changes to this file are for testing the bug I fixed above (the condition I added in line 363 of file `reactor.go`). If you remove that condition, the new UT fails.

Reply:
> this could be a parameter to allow creating genesis of both kinds.

Genesis creation needs a good refactor, but I chose not to do it for now.